TLM/TLM/Util/DetailLogger.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using CSUtil.Commons;
+
+namespace TrafficManager.Util {
+    internal static class DetailLogger {
+
+        public delegate void LogDelegate(params object[] lines);
+
+        [Conditional("DEBUG")]
+        public static void LogDebug(string heading, params object[] lines) {
+            var entry = new StringBuilder();
+
+            if (heading != null) {
+                entry.Append(heading);
+                entry.Append(' ');
+            }
+            string newline = string.Empty;
+
+            foreach (var o in lines) {
+                entry.Append(newline);
+                newline = "\n";
+                if (o is string) {
+                    entry.Append(o);
+                } else {
+                    BuildBLock(o);
+                }
+            }
+            Log._Debug(entry.ToString());
+
+            void BuildBLock(object o) {
```

Comment:
```suggestion
             static void BuildBLock(object o) {
```

Reply:
I disagree. This just copies the reference into the new stack frame, and doesn't really seem to add any value.